shared/common-adapters/text.meta.desktop.js

```diff
@@ -0,0 +1,260 @@
+// @flow
+import {globalStyles, globalColors} from '../styles'
+
+import type {MetaType, TextType, Background} from './text'
+
+function defaultColor (backgroundMode: ?Background) {
+  if (!backgroundMode) {
+    backgroundMode = 'Normal'
+  }
+
+  return {
+    'Normal': globalColors.white,
+    'Announcements': globalColors.white,
+    'Success': globalColors.white,
+    'Information': globalColors.brown_60,
+    'HighRisk': globalColors.white,
+    'Documentation': globalColors.white,
+    'Terminal': globalColors.white,
+  }[backgroundMode]
+}
+
+function lineClamp (lines: number): Object {
+  return {
+    overflow: 'hidden',
+    display: '-webkit-box',
+    textOverflow: 'ellipsis',
+    WebkitBoxOrient: 'vertical',
+    WebkitLineClamp: lines,
+  }
+}
+
+function fontSizeToSizeStyle (fontSize: number): ?Object {
+  const height = {
+    '24': 28,
+    '16': 20,
+    '14': 18,
+    '13': 17,
+    '11': 15,
+  }[String(fontSize)]
+
+  const lineHeight = height ? `${height}px` : null
+  return {
+    fontSize,
+    lineHeight,
+  }
+}
+
+const _blackNormalWhiteTerminal = {
+  'Normal': globalColors.black_75,
+  'Terminal': globalColors.white,
+}
+
+const _blueLink = {
+  'Normal': globalColors.blue,
+}
+
+const metaData: {[key: TextType]: MetaType} = {
+  // Header
+  'HeaderBig': {
```

Comment:
There's some duplication here but i think it's simpler to understand vs a lot of indirection and overloading of things